pkg/pod/pod.go

```diff
@@ -174,6 +179,13 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1.TaskRun, taskSpec v1.Ta
 	enableKeepPodOnCancel := featureFlags.EnableKeepPodOnCancel
 	setSecurityContext := config.FromContextOrDefaults(ctx).FeatureFlags.SetSecurityContext
 
+	// Kubernetes Version
+	dc := b.KubeClient.Discovery()
+	sv, err := dc.ServerVersion()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I am not very familiar with the `DiscoveryInterface()`. I tried to find more about it but could not find determine whether it results in an additional API call. Since we have a feature flag for this functionality. I recommend placing it under the `if` condition that checks the feature flag.


```
	if config.FromContextOrDefaults(ctx).FeatureFlags.EnableKubernetesSidecar {
		// Kubernetes Version
		dc := b.KubeClient.Discovery()
		sv, err := dc.ServerVersion()
		if err != nil {
			return nil, err
		}
		// Check if current k8s version is less than 1.29
		// Since Kubernetes Major version cannot be 0 and if it's 2 then sidecar will be in
		// we are only concerned about major version 1 and if the minor is less than 29 then
		// we need to do the current logic
		svMinorInt, _ := strconv.Atoi(sv.Minor)
		svMajorInt, _ := strconv.Atoi(sv.Major)
		if svMajorInt >= 1 && svMinorInt >= SidecarK8sMinorVersionCheck {
			// Add RestartPolicy and Merge into initContainer
			for i := range sidecarContainers {
				sc := &sidecarContainers[i]
				always := corev1.ContainerRestartPolicyAlways
				sc.RestartPolicy = &always
				sc.Name = names.SimpleNameGenerator.RestrictLength(fmt.Sprintf("%v%v", sidecarPrefix, sc.Name))
				mergedPodInitContainers = append(mergedPodInitContainers, *sc)
			}
		}
	} else {
		// Merge sidecar containers with step containers.
		for _, sc := range sidecarContainers {
			sc.Name = names.SimpleNameGenerator.RestrictLength(fmt.Sprintf("%v%v", sidecarPrefix, sc.Name))
			mergedPodContainers = append(mergedPodContainers, sc)
		}
	}
```

Reply:
I did slightly different logic to handle the case where you set the enable-kubernetes-sidecar to 'true' but you're running on a kubernetes version less than 1.29